itests/eth_balance_test.go

```diff
@@ -0,0 +1,71 @@
+package itests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+	"github.com/filecoin-project/lotus/itests/kit"
+)
+
+func TestEthGetBalanceExistingF4address(t *testing.T) {
+	blockTime := 100 * time.Millisecond
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+	ens.InterconnectAll().BeginMining(blockTime)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	_, ethAddr, deployer := client.EVM().NewAccount()
+
+	fundAmount := types.FromFil(10)
+	// send some funds to the f410 address
+	kit.SendFunds(ctx, t, client, deployer, fundAmount)
+
+	balance, err := client.EthGetBalance(ctx, ethAddr, "latest")
+	require.NoError(t, err)
+	require.Equal(t, balance, ethtypes.EthBigInt{Int: fundAmount.Int})
+}
+
+func TestEthGetBalanceNonExistentF4address(t *testing.T) {
+	blockTime := 100 * time.Millisecond
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+	ens.InterconnectAll().BeginMining(blockTime)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	_, ethAddr, _ := client.EVM().NewAccount()
+
+	balance, err := client.EthGetBalance(ctx, ethAddr, "latest")
```

Comment:
Grr, strings bad. (not your fault, but feel free to add an enum if you're inspired)

Reply:
This usage is fine. Its how the ethereum API is defined
https://docs.alchemy.com/reference/eth-getbalance

Reply:
Yeah, this is unfortunately literally the Eth API spec.